packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -57,6 +94,128 @@ Future<Depfile> copyAssets(Environment environment, Directory outputDirectory) a
   return Depfile(inputs, outputs);
 }
 
+/// Returns a map of [FontSubsetData] keyed by relative path.
+Future<Map<String, FontSubsetData>> _getIconData(Environment environment, DevFSStringContent fontManifest) async {
+  final File appDill = environment.buildDir.childFile('app.dill');
+
+  final File constFinder = globals.fs.file(globals.artifacts.getArtifactPath(Artifact.constFinder));
+  final File dart = globals.fs.file(globals.artifacts.getArtifactPath(Artifact.engineDartBinary));
+
+  final Map<String, List<int>> iconData = await _findConstants(dart, constFinder, appDill);
+  final Set<String> familyKeys = iconData.keys.toSet();
+
+  final Map<String, String> fonts = await _parseFontJson(fontManifest.string, familyKeys);
+
+  if (fonts.length != iconData.length) {
+    throwToolExit('Expected to find fonts for ${iconData.keys}, but found ${fonts.keys}.');
+  }
+
+  final Map<String, FontSubsetData> result = <String, FontSubsetData>{};
+  for (final MapEntry<String, String> entry in fonts.entries) {
+    result[entry.value] = FontSubsetData(
+      family: entry.key,
+      relativePath: entry.value,
+      codePoints: iconData[entry.key],
+    );
+  }
+  globals.printTrace('Found iconData: $result');
+  return result;
+}
+
+/// Calls font-subset, which transforms the [inputPath] font file to a subsetted
+/// version containing only the [FontSubsetData.codePoints] at [outputPath].
+Future<void> _subsetFont({
+  @required Environment environment,
+  @required String inputPath,
+  @required String outputPath,
+  @required FontSubsetData fontSubsetData,
+}) async {
+  final File fontSubset = globals.fs.file(globals.artifacts.getArtifactPath(Artifact.fontSubset));
+
+  final List<String> cmd = <String>[
+    fontSubset.path,
+    outputPath,
+    inputPath,
+  ];
+  globals.printTrace('Starting font-subset: ${cmd.join(' ')}...');
```

Comment:
I would printTrace less here since the command runs in a pool with concurrency they'll be stacked up in ways that don't make sense